src/test/modelmanager_test.cpp

```diff
@@ -526,6 +527,59 @@ TEST(ModelManager, ConfigReloadingShouldAddNewModel) {
     modelMock.reset();
 }
 
+struct CNLIMWrapperMock : public ovms::CNLIMWrapper {
+    inline static int deinitializeSum = 0;
+
+public:
+    CNLIMWrapperMock(void* CNLIM, ovms::deinitialize_fn deinitialize) :
+        ovms::CNLIMWrapper(CNLIM, deinitialize) {}
+
+    ~CNLIMWrapperMock() {
+        deinitializeSum += deinitialize(ptr);
+    }
+};
+
+TEST(ModelManager, ConfigReloadShouldCleanupResources) {
+    ResourcesAccessModelManager manager;
+    manager.startCleaner();
+    ASSERT_EQ(manager.getResourcesSize(), 0);
+
+    // Reset mocked wrapper deinitializeSum
+    CNLIMWrapperMock::deinitializeSum = 0;
+
+    int num1 = 1;
+    int num2 = 19;
+    int num3 = 11;
+    {
+        std::shared_ptr<CNLIMWrapperMock> ptr1 = std::make_shared<CNLIMWrapperMock>(&num1, [](void* ptr) {
+            int* number = static_cast<int*>(ptr);
+            return *number;
+        });
+        std::shared_ptr<CNLIMWrapperMock> ptr2 = std::make_shared<CNLIMWrapperMock>(&num2, [](void* ptr) {
+            int* number = static_cast<int*>(ptr);
+            return *number;
+        });
+        std::shared_ptr<CNLIMWrapperMock> ptr3 = std::make_shared<CNLIMWrapperMock>(&num3, [](void* ptr) {
+            int* number = static_cast<int*>(ptr);
+            return *number;
+        });
+
+        manager.addResourceToCleaner(ptr1);
+        manager.addResourceToCleaner(ptr2);
+        manager.addResourceToCleaner(std::move(ptr3));
+        ASSERT_EQ(manager.getResourcesSize(), 3);
+
+        waitForOVMSResourcesCleanup(manager);
+        ASSERT_EQ(manager.getResourcesSize(), 2);
+        ASSERT_EQ(CNLIMWrapperMock::deinitializeSum, num3);
+    }
+    waitForOVMSResourcesCleanup(manager);
+    ASSERT_EQ(manager.getResourcesSize(), 0);
```

Comment:
We should have tests for cleaner routine that:
1. No sequences cleanup. Resources are cleaned up in demanded time (eg if we set timeout for 5s we expect the resources to be removed after at most 6s).
2. With sequences cleanup & both timeouts different and prefereably prime numbers eg, 5ms, 7ms, verify that both resources are cleaned up within reasonable timeframe - 6ms & 8ms respectively